`src/System/Reflection/Assembly.CoreCLR.cs(40,20): error CS0246: The type or namespace name 'FileNotFoundException' could not be found (are you missing a using directive or an assembly reference?)`

--- a/src/binder/assemblyname.cpp
+++ b/src/binder/assemblyname.cpp
@@ -486,6 +486,15 @@ Exit:
         {
             dwUseIdentityFlags &= ~AssemblyIdentity::IDENTITY_FLAG_CONTENT_TYPE;
         }
+        if ((dwIncludeFlags & INCLUDE_PUBLIC_KEY_TOKEN) == 0)
+        {
+            dwUseIdentityFlags &= ~AssemblyIdentity::IDENTITY_FLAG_PUBLIC_KEY;
+            dwUseIdentityFlags &= ~AssemblyIdentity::IDENTITY_FLAG_PUBLIC_KEY_TOKEN;
+        }
+        if ((dwIncludeFlags & EXCLUDE_CULTURE) != 0)
+        {
+            dwUseIdentityFlags &= ~AssemblyIdentity::IDENTITY_FLAG_CULTURE;
+        }
 
         dwHash ^= static_cast<DWORD>(HashCaseInsensitive(GetSimpleName()));
         dwHash = _rotl(dwHash, 4);

--- a/tests/src/Loader/ContextualReflection/ContextualReflection.cs
+++ b/tests/src/Loader/ContextualReflection/ContextualReflection.cs
@@ -751,7 +751,6 @@ namespace ContextualReflectionTest
             TestMockAssemblyThrows();
         }
 
-        [MethodImplAttribute(MethodImplOptions.NoInlining)]
         public void RunTestsIsolated()
         {
             VerifyIsolationAlc();

--- a/tests/src/Loader/ContextualReflection/ContextualReflection.cs
+++ b/tests/src/Loader/ContextualReflection/ContextualReflection.cs
@@ -321,6 +321,9 @@ namespace ContextualReflectionTest
         {
             TestAssemblyLoad(isolated, (string assemblyName) => Assembly.Load(assemblyName));
             TestAssemblyLoad(isolated, (string assemblyName) => Assembly.Load(new AssemblyName(assemblyName)));
+#pragma warning disable 618
+            TestAssemblyLoad(isolated, (string assemblyName) => Assembly.LoadWithPartialName(assemblyName));
+#pragma warning restore 618
         }
 
         void TestAssemblyLoad(bool isolated, Func<string, Assembly> assemblyLoad)

--- a/tests/src/Loader/ContextualReflection/ContextualReflectionDependency.cs
+++ b/tests/src/Loader/ContextualReflection/ContextualReflectionDependency.cs
@@ -16,7 +16,6 @@ namespace ContextualReflectionTest
         Assembly alcAssembly { get; }
         Type alcProgramType { get; }
         IProgram alcProgramInstance { get; }
-        [MethodImplAttribute(MethodImplOptions.NoInlining)]
         void RunTestsIsolated();
     }
 

